docs/code/EntryPoints.md

```diff
@@ -0,0 +1,295 @@
+﻿# Entry Points And Helper Classes 
+
+## Overview
+
+An 'entry point', is a representation of a ML.NET type in JSON format. Entry points are used to serialize and deserialize an ML.NET type in JSON. 
+It is also the recommended way to interface with other languages. 
+Defined based on entry points, experiments are entry points DAGs, and respectively, entry points are experiment graph nodes.
+That's why through the documentaiton, we also refer to them as 'entry points nodes'.
+The graph 'variables', the various values of the experiment graph JSON properties serve to describe the relationship between the entry point nodes. 
+The 'variables' are therefore the edges of the DAG. 
+
+All of ML.NET entry points are described by their manifest. The manifest is another JSON object that documents and describes the structure of an entry points. 
+Manifests are referenced to understand what an entry point does, and how it should be constructed, in a graph.  
+
+This document briefly describes the structure of the entry points, the structure of an entry point manifest, and mentions the ML.NET classes that help construct an entry point
+graph.
+
+## EntryPoint manifest - the definition of an entry point
+
+An example of an entry point manifest object, specifically for the MissingValueIndicator transform, is:
+
+```javascript
```

Comment:
This is how it is actually written out, but I wonder if we could just format it a bit to make it a bit more tolerable. The document is dominated by this ~180 line monstrosity. I think it could be improved significantly by just deleting a bunch of whitespace... so for example if the stuff from lines 40 through 65, we coudl 

```javascript
"Values": ["I1", "U1", "I2", "U2", "I4", "U4", "I8", "U8",
    "R4", "Num", R8", "TX", "Text", "TXT", "BL", "Bool",
    "TimeSpan", "TS", "DT", DateTime", "DZ", "DateTimeZone",
    "UG", "U16"]
```